File: c/eliuds-eggs/eliuds_eggs.c
#include "eliuds_eggs.h"

#include <string.h>
#include <stdio.h>

unsigned int egg_count(unsigned int input) {
	unsigned int lines = 0;
	while(input > 0) {
		lines += input & 1;
		input >>= 1;
	}
	return lines;
}

File: php/eliuds-eggs/EliudsEggs.php
<?php

declare(strict_types=1);

class EliudsEggs
{
    public function eggCount(int $displayValue): int
    {
	    $ones = 0;
	    // throw new \BadMethodCallException("Implement the eggCount function");
	    while ($displayValue > 0) {
		$ones += $displayValue & 1;
		$displayValue >>= 1;
	    }
	    return $ones;
    }
}
